From 1429f44472e96c0d5fd1e4b8686d193030b44be4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 7 Sep 2021 08:32:27 +0200
Subject: [PATCH edk2-test 1/1] uefi-sct/SctPkg: unsupported
 TEXT_INPUT_EX.SetState

REF: https://bugzilla.tianocore.org/show_bug.cgi?id=3390

According to the UEFI specification
EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL.SetState() may return EFI_UNSUPPORTED if
"the device does not support the ability to have its state set."

BBTestReadKeyStrokeExFunctionAutoTestCheckpoint1() must not report an error
in this case.

Fixes: e9c21711c17 ("uefi-sct/SctPkg:Add checkpoint of ReadKeyStrokeEx Toggle state")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 .../BlackBoxTest/SimpleTextInputExBBTestFunction.c             | 3 +++
 .../BlackBoxTest/SimpleTextInputExBBTestFunction.c             | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
index 3ae40ee3b3ed..c9eaf408ab02 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
@@ -1182,6 +1182,9 @@ BBTestReadKeyStrokeExFunctionAutoTestCheckpoint1 (
                                 SimpleTextInputEx,
                                 &State
                                 );
+  if (Status == EFI_UNSUPPORTED) {
+    return EFI_UNSUPPORTED;
+  }
 
   if (EFI_ERROR(Status)) {
     StandardLib->RecordAssertion (
diff --git a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
index 7dfd5dea6269..eddce4d3640d 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/IHV/Protocol/SimpleTextInputEx/BlackBoxTest/SimpleTextInputExBBTestFunction.c
@@ -1182,6 +1182,9 @@ BBTestReadKeyStrokeExFunctionAutoTestCheckpoint1 (
                                 SimpleTextInputEx,
                                 &State
                                 );
+  if (Status == EFI_UNSUPPORTED) {
+    return EFI_UNSUPPORTED;
+  }
 
   if (EFI_ERROR(Status)) {
     StandardLib->RecordAssertion (
-- 
2.30.2

